src/psyclone/tests/domain/lfric/kernel/inter_grid_arg_metadata_test.py

```diff
@@ -0,0 +1,229 @@
+# Author R. W. Ford, STFC Daresbury Lab
+
+'''Module containing tests for the InterGridArgMetadata class.
+
+'''
+import pytest
+
+from fparser.two import Fortran2003
+
+from psyclone.domain.lfric.kernel import InterGridArgMetadata
+
+
+def test_create():
+    '''Test that an instance of InterGridArgMetadata can be created
+    successfully.
+
+    '''
+    inter_grid_arg = InterGridArgMetadata(
+        "GH_REAL", "GH_READ", "W0", "GH_FINE")
+    assert isinstance(inter_grid_arg, InterGridArgMetadata)
+    assert inter_grid_arg.form == "gh_field"
+    assert inter_grid_arg._datatype == "gh_real"
+    assert inter_grid_arg._access == "gh_read"
+    assert inter_grid_arg._function_space == "w0"
+    assert inter_grid_arg._mesh_arg == "gh_fine"
+
+
+def test_init_invalid():
+    '''Test that an invalid mesh supplied to the constructor
+    raises the expected exception.
+
+    '''
+    with pytest.raises(ValueError) as info:
+        _ = InterGridArgMetadata("GH_REAL", "GH_READ", "W0", "invalid")
+    assert ("The mesh_arg metadata should be a recognised value (one of "
+            "['gh_coarse', 'gh_fine']) but found 'invalid'."
+            in str(info.value))
+
+
+def test_create_from_fparser2():
+    '''Test that the create_from_fparser2 static method works as
+    expected. Test that all relevant check and get methods are called
+    by raising exceptions within them, as well as checking for valid
+    input.
+
+    '''
+    with pytest.raises(TypeError) as info:
+        _ = InterGridArgMetadata.create_from_fparser2("hello")
+    assert ("Expected kernel metadata to be encoded as an fparser2 "
+            "Structure_Constructor object but found type 'str' with value "
+            "'hello'." in str(info.value))
+
+    fparser2_tree = InterGridArgMetadata.create_fparser2(
+        "hello(x=a)", encoding=Fortran2003.Structure_Constructor)
+    with pytest.raises(ValueError) as info:
+        _ = InterGridArgMetadata.create_from_fparser2(fparser2_tree)
+    assert ("Expected kernel metadata to have the name 'arg_type' "
+            "and be in the form 'arg_type(...)', but found 'hello(x = a)'."
+            in str(info.value))
+
+    fparser2_tree = InterGridArgMetadata.create_fparser2(
+        "arg_type(x=a)", encoding=Fortran2003.Structure_Constructor)
+    with pytest.raises(ValueError) as info:
+        _ = InterGridArgMetadata.create_from_fparser2(fparser2_tree)
+    assert ("Expected kernel metadata to have 5 arguments, but "
+            "found 1 in 'arg_type(x = a)'." in str(info.value))
+
+    fparser2_tree = InterGridArgMetadata.create_fparser2(
+        "arg_type(GH_FEELED, GH_REAL, GH_READ, W0, mesh_arg=GH_COARSE)",
+        encoding=Fortran2003.Structure_Constructor)
+    with pytest.raises(ValueError) as info:
+        _ = InterGridArgMetadata.create_from_fparser2(fparser2_tree)
+    assert ("InterGrids should have gh_field as their first metadata "
```

Comment:
I noticed this in previous tests but didn't comment there - I don't like "InterGrids" as they are not a thing. Perhaps "Metadata for inter-grid kernel arguments should have ....."